woocommerce/woocommerce-ios
# [Custom Amounts M1] Add custom amount to order as fee - Business Logic

Closes: #10925 


## Description

With this PR we implement the business logic for adding a fee/remove to an order. Since before we only supported one fee per order, we were always replacing that unique fee. Now that we can have multiple fees (custom amounts) linked to an order, we need these new methods.

You can see how we left the previous code, due to the fact that we still want that logic in production. Once we enable this new functionality by removing the feature flag, we will remove also the old logic.

## Testing instructions


We do not show the added custom amounts in order creation yet -it will be part of the next PR-, but there are a couple of ways to verify that adding a custom amount as a fee works effectively:

1. Go to Orders
2. Tap on + Add Custom Amount
3. Add a custom amount with a name
4. See how the Fees field in the Payment section is updated (this will be removed in the future but for now it helps to test)
5. Repeat the process
6. See how the Fees field in the Payment section is updated. same for total.
7. Tap on Create to create the order
8. Go to wp-admin to verify that the added custom amounts, checking their names as well 

Please check that the previous fee feature -adding a fee from the + Add fee field - works as before.

## Screenshots


### Recording

https://github.com/woocommerce/woocommerce-ios/assets/1864060/6f2b20cb-a13f-4d32-88ae-baf87f98bb1c

### Order in wp-admin

<img width="1416" alt="Screenshot 2023-10-23 at 14 29 57" src="https://github.com/woocommerce/woocommerce-ios/assets/1864060/0bfd6bd2-ebb4-4a87-be13-f40c24ce7695">

---
